### ISSUE TITLE
False negative on OutdatedDocumentation around private property

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#6371]
## Expected Behavior
```kotlin
@Test
fun `should report when param which is private property is documented as property`() {
    val code = """
        /**
         * Doc
         * @property a desc
         */
        class A(
            private val a: String,
        )
    """.trimIndent()
    assertThat(subject.compileAndLint(code)).hasSize(1)
}
```
The above TC should pass. As `a` is private property and hence should only be documented as `@param`

## Observed Behavior
The above TC is failing

## Steps to Reproduce
Run the TC against the main code

## Context
While seeing the issue at https://github.com/detekt/detekt/issues/6362, the above TC failed

## Your Environment

* Version of detekt used: latest main
* Version of Gradle used (if applicable): detekt repo
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: NA
* Link to your project (if it's a public repository): detekt repo
